baby-gru/src/components/list/MoorhenCarbohydrateList.tsx

```diff
@@ -0,0 +1,62 @@
+import { useEffect, useRef, useState } from "react";
+import { Row, Col } from "react-bootstrap";
+import { moorhen } from "../../types/moorhen";
+import { webGL } from "../../types/mgWebGL";
+import { useSelector } from "react-redux";
+import { getLigandSVG } from "../../utils/MoorhenUtils";
+import { MoorhenCarbohydrateCard } from "../card/MoorhenCarbohydrateCard";
+import {PrivateerResultsEntry} from "../../types/privateer";
+
+export const MoorhenCarbohydrateList = (props: {
+    setBusy?: React.Dispatch<React.SetStateAction<boolean>>;
+    commandCentre: React.RefObject<moorhen.CommandCentre>;
+    molecule: moorhen.Molecule;
+    glRef: React.RefObject<webGL.MGWebGL>; 
+    height?: number | string;
+}) => {
+
+    const isDark = useSelector((state: moorhen.State) => state.sceneSettings.isDark)
+
+    const [carbohydrateList, setCarbohydrateList] = useState<PrivateerResultsEntry[] | null>(null)
+
+    const validate = async () => {
+        if (props.molecule) {
+            console.log(props.molecule)
+            const privateerResult = await props.commandCentre.current.cootCommand({
+                command: 'privateer_validate',
+                commandArgs: [props.molecule.molNo],
+                returnType: 'privateer_results'
+            }, false)
+
+            const privateerData: PrivateerResultsEntry[] = privateerResult.data.result.result;
+            console.log(privateerResult)
+            setCarbohydrateList(privateerData)
+        }
+    }
+
+    useEffect(() => {
+        validate()
+    }, [props.molecule])
```

Comment:
Probably a better dependency for this is `newCootCommandAlert`. It's a state in the redux store that informs of changes to a molecule. You can declare it like this:
```js
const newCootCommandAlert = useSelector((state: moorhen.State) => state.generalStates.newCootCommandAlert)
```
And then simply use it here in the dependency array instead of `props.molecule`